#! /bin/sh /usr/share/dpatch/dpatch-run
## 02_CVS-20060920.dpatch by  <jeremie.corbier@resel.enst-bretagne.fr>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Update to CVS 20060920

@DPATCH@

--- orig/CHANGES	2006/07/30 10:24:19	1.15
+++ dest/CHANGES	2006/09/20 08:32:31	1.20
@@ -1,3 +1,22 @@
+2006-09-20  SUZUKI, Shinsuke <suz@kame.net>
+	* common.c: fixed a bug that dhcp6s inserts SIP server addresses into
+	  DNS server address option (Bug-ID 1561202).  (degrade from 2006-07-30)
+	* common.c: fixed a bug that status-code option might include a unexpected
+	  garbage value.  (Bug-ID 1561202)
+	* dhcp6s.conf.5, dhcp6s.conf.sample: you need to provide a preferred-lifetime
+	  for each address-prefix.  (Bug-ID 1558811)
+
+2006-09-20  STEVANT, Bruno <bruno.stevant@enst-bretagne.fr>
+	* lease.c: Fix a memory violation in lease management.
+
+	--- 20060902 tar ball released ---
+2006-09-02  SUZUKI, Shinsuke <suz@kame.net>
+	* added a configure check routine to automatically detect a 
+	  difference in TAILQ_FOREACH_REVERSE macros (Suggested by Brute
+	  STEVANT)
+	* fixed a bug that dhcp6c cannot advertise an Option-Request-Option.
+	* fixed a compilation failure in freebsd4
+
 2006-07-30  SUZUKI, Shinsuke <suz@kame.net>
 	* supported the following options:
 		NIS server option, NIS domain option (RFC3898)
--- orig/common.c	2006/07/30 10:24:19	1.4
+++ dest/common.c	2006/09/20 08:22:24	1.6
@@ -2166,7 +2166,7 @@
 	     stcode = TAILQ_NEXT(stcode, link)) {
 		u_int16_t code;
 
-		code = htons(stcode->val_num);
+		code = htons(stcode->val_num16);
 		if (copy_option(DH6OPT_STATUS_CODE, sizeof(code), &code, &p,
 		    optep, &len) != 0) {
 			goto fail;
@@ -2222,7 +2222,7 @@
 	    &p, optep, &len) != 0)
 		goto fail;
 
-	if (dhcp6_set_addr(DH6OPT_DNS, &optinfo->sip_list,
+	if (dhcp6_set_addr(DH6OPT_DNS, &optinfo->dns_list,
 	    &p, optep, &len) != 0)
 		goto fail;
 
--- orig/dhcp6s.conf.5	2006/07/30 10:24:19	1.6
+++ dest/dhcp6s.conf.5	2006/09/20 08:32:31	1.7
@@ -215,7 +215,7 @@
 advertise messages.
 The preference value must be a decimal integer and be between 0 and
 255 (inclusive.)
-.It Ic address-pool Ar pool Op Ar pltime Op Ar vltime ;
+.It Ic address-pool Ar pool Ar pltime Op Ar vltime ;
 This statement assigns an address pool
 .Ar pool
 to the interface. When
--- orig/dhcp6s.conf.sample	2006/01/26 07:19:54	1.2
+++ dest/dhcp6s.conf.sample	2006/09/20 08:32:31	1.3
@@ -10,12 +10,12 @@
 };
 
 # The followings are a sample configuration to provide an IPv6 address
-# from an address pool 2001:db8:1:2::1000-2000.
+# from an address pool 2001:db8:1:2::1000-2000 for 3600[s].
 # Note. You have to send an RA with A-bit off to fxp0; otherwise
 # a client cannot be sure about the prefix-length and the default router.
 
 interface fxp0 {
-	address-pool pool1;
+	address-pool pool1 3600;
 };
 
 pool pool1 {
--- orig/lease.c	2006/09/02 08:19:11	1.4
+++ dest/lease.c	2006/09/20 07:57:15	1.5
@@ -217,6 +217,8 @@
 		while (!LIST_EMPTY(&table->table[i])) {
 			struct hash_entry *entry = LIST_FIRST(&table->table[i]);
 			LIST_REMOVE(entry, list);
+			if (entry->val)
+				free(entry->val);
 			free(entry);
 		}
 	}
@@ -237,10 +239,14 @@
 		return (-1);
 	}
 
-	if ((entry = malloc(sizeof(*entry) + size)) == NULL) {
+	if ((entry = malloc(sizeof(*entry))) == NULL) {
 		return (-1);
 	}
 	memset(entry, 0, sizeof(*entry));
+
+	if ((entry->val = malloc(size)) == NULL) {
+		return (-1);
+	}
 	memcpy(entry->val, val, size);
 
 	i = table->hash(val) % table->size;
@@ -265,6 +271,8 @@
 	}
 
 	LIST_REMOVE(entry, list);
+	if (entry->val)
+		free(entry->val);
 	free(entry);
 
 	return (0);
